### ISSUE TITLE
HTTPS connection to HEC

### ISSUE THREAD

**Issue by npeyrussie** [splunk/splunk-connect-for-kubernetes#705]
Hello,

I am trying to configure Splunk Connect for Kubernetes to capture a k8s cluster application logs.

I have problems when configuring https connection to HEC.

On the Heavy Forwarder, I have configured a ServerCert Configuration, which has been signed by our Company Authority.
Configuration is OK and if I openssl on HEC port I get the right certificate.

Then, on Splunk Connect for Kubernetes Helm, if I configure https :
```
  splunk:
    hec:
      # host is required and should be provided by user
      host: hostname.domain.com
      # token is required and should be provided by user
      token: MY-HEC-TOKEN
      # protocol has two options: "http" and "https", default is "https"
      # For self signed certificate leave this field blank
      protocol: https
 ```

When deploying, I see the following logs on Heavy Forwarder splunkd.log : 
```
01-25-2022 09:37:16.729 +0100 WARN  SSLCommon [1235867 HttpInputServerDataThread] - Received fatal SSL3 alert. ssl_state='SSLv3 read client key exchange A', alert_description='unknown CA'.
01-25-2022 09:37:16.729 +0100 WARN  HttpListener [1235867 HttpInputServerDataThread] - Socket error from 10.8.199.195:55608 while idling: error:14094418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca - please check the output of the `openssl verify` command for the certificates involved; note that if certificate verification is enabled (requireClientCert or sslVerifyServerCert set to "true"), the CA certificate and the server certificate should not have the same Common Name.
 ```
I have to add `insecureSSL: true` to get the connection working and see logs on Indexer.

But, If I activate HTTPS connection, I do not want it to be insecure ^^.

I am a bit confused about Splunk Connect 4 Kubernetes configuration.

Can I use : 
```
splunk:
  # Configurations for HEC (HTTP Event Collector)
  hec:
    # The PEM-format CA certificate file.
    # NOTE: The content of the file itself should be used here, not the file path.
    #       The file will be stored as a secret in kubernetes.
    caFile:
 ```
To configure ma Company CA ? I tried but still got the same error.

Or are keys `clientCert`, `clientKey` and `CaFile` only used for mTLS configuration ?

Then how can I make Splunk for Kubernetes trust my Company CA ?

Thank you in advance for your answers.

Regards.

Nicolas.

**Comment by harshit-splunk** [comment]
Hey @npeyrussie, have you checked `openssl verify` command? 

> To configure ma Company CA ? I tried but still got the same error.

Can you please specify how did you configured  company CA?

**Comment by npeyrussie** [comment]
Hello @harshit-splunk  and sorry for the delay.

I configured my Company CA as this (here is a fake value) : 
```
  splunk:
    hec:
      insecureSSL: true
      caFile: |
        -----BEGIN CERTIFICATE-----
        MIIF+jCCBOKgAwIBAgIKYBag0QABAAAAGDANBgkqhkiG9w0BAQsFADAYMRYwFAYD
        VQQDEw1NQUNJRiBST09UIENBMB4XDTIwMDUyNTEyMjA0N1oXDTQwMDUyNTEyMzA0
        N1owgYIxCzAJBgNVBAYTAkZSMRswGQYDVQQIExJOb3V2ZWxsZSBBcXVpdGFpbmUx
        DjAMBgNVBAcTBU5pb3J0MQ4wDAYDVQQKEwVNQUNJRjEbMBkGA1UECwwSRERTSUdf
        QUlTX1NlY3VyaXRlMRkwFwYDVQQDExBNQUNJRiBTdWIgQ0EgTDFGMIICIjANBgkq
        hkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAwkN6PgZE0N+ff5wcSO2+4a4tTrjtDkm2
        ZXDcr9MzkM6p2u2VxCFkZMdVYx7vjzHP9IKP45Wdt6unqqQsldCFS0+NNzk6zglb
        Ft8t/vOFMIFIyNA0VEQE41CYI72sYMwrPiEyvmS6tIHk+JBU1NPfj25Py6e/2IIW
        j6VBmGpEAR4BwytUmQIIEu68mYPy1GGJP48PVxJFntxng+S+TTlBFvqv6DXjm7je
        q+eSlE4Q/ce5UO/YVaFmZjqelS9NHg/X+BBl/AWGG0vRY0NbPh9P1X97tYLUXzVa
        xJ7jFbfmMFudlkEl297XmdzG7QkZt09VBwWgl3h2lCqxTU9ZK6WVti3uDOqvLxxv
        NXAEm4APwFNzlvNM/Qwy4OM50OIbcTn9Bv7LkB5+0ZGK535bjw12Nk+tre14E+QC
        tEaJI9QE/AbhPlom05KdbRuIo4sSwUtedK5Lq9OgAAG7xKf0e8ED/hvY5sFijRJV
        F4gcdrCcrRdP/o9h8jLqMpRFGNmAq+WZlo6VCVnIPFUFIKx4fHTR4ENN0i5FyLOK
        dezuHQdKklSnGJdoUEvLBtdPJ6IQ3t1iPW11O7xoLzd2s1EKzSuRjG8ucUHTNMKo
        zNDbjMnrxA8Oxdu+SYmx5v7gAcT8h0qtljy2RuL+azcu0F0F/8IEgOaHmiEIr4A0
        QAwGB/DBCfMCAwEAAaOCAdkwggHVMB0GA1UdDgQWBBQ1q1VJVzKJ2SDgkbLXAIN0
        8pPeiDALBgNVHQ8EBAMCAeYwEgYDVR0TAQH/BAgwBgEB/wIBADA9BgNVHREENjA0
        ghBNQUNJRiBTdWIgQ0EgTDFGgghOQ0xSMDAwMoEWY3liZXJzZWN1cml0ZUBtYWNp
        Zi5mcjAfBgNVHSMEGDAWgBTavla7vFOARDyob3oAufRxFUEwZDBPBgNVHR8ESDBG
        MESgQqBAhj5odHRwOi8vcGtpLmdyb3VwZS5hZC5tYWNpZi5mci9DZXJ0RW5yb2xs
        L01BQ0lGJTIwUk9PVCUyMENBLmNybDCBxgYIKwYBBQUHAQEEgbkwgbYwgbMGCCsG
        AQUFBzAChoGmbGRhcDovLy9DTj1NQUNJRiUyMFJPT1QlMjBDQSxDTj1BSUEsQ049
        UHVibGljJTIwS2V5JTIwU2VydmljZXMsQ049U2VydmljZXMsQ049Q29uZmlndXJh
        dGlvbixEQz1BRCxEQz1NQUNJRixEQz1GUj9jQUNlcnRpZmljYXRlP2Jhc2U/b2Jq
        ZWN0Q2xhc3M9Y2VydGlmaWNhdGlvbkF1dGhvcml0eTAZBgkrBgEEAYI3FAIEDB4K
        AFMAdQBiAEMAQTANBgkqhkiG9w0BAQsFAAOCAQEAny2l3f/nk2xxokP/l6GJu6Ce
        tFaS1W369sbdIJqwN3YZ3oSDZPlDKcogBlZD8gR1n+UuH+tDk+dowLmGwTlGNj8w
        VIx9yKTJmFMit+VIyto7dmbox4jgnc79cQkXr4ZUOuhsU6Srhag2PGmOcp4wqvSx
        wljDl4YiRI+AwJDct6HpEFOllClzyvMQ1/P0tiSpsgfu4Und+91ourqgaK6OhJAi
        aR8pSVbi8oUbKNHAvBFVxwTHAwZ1CRS+NsQMA+rXZPV2XsT7Ai8V0TM65/WvdqV6
        EAbD8hmhdGnXvO9I4kW6/f4diBUj+drl0R5FFKcAgl9j2KnDsy8VFEhfYxVDEg==
        -----END CERTIFICATE-----
```

**Comment by harshit-splunk** [comment]
Hello @npeyrussie, sorry for the delayed response. I am not sure why you are receiving `unknown CA` logs in heavy forwarder. 

> Is insecureSSL: true refering to HTTPS connection To HEC or to mTLS connection with it ?

No, it is used when you are using self-signed certificates. When it is enabled, it will not verify server's cert.

There was an issue with SCK in reading certificate from values.yaml file. https://github.com/splunk/splunk-connect-for-kubernetes/pull/725 should fix that. Once it is merged, disable insecureSSL and provide root CA certificate as `caFile`.

**Comment by npeyrussie** [comment]
Hello @harshit-splunk , sorry for the delay.
I just tried this morning, applying the commit diff to splunk-kubernetes-logging/templates/secret.yaml file.

Now it seems to be OK with my CA file.
I could delete the `insecureSSL: true` from my values file and I can find logs on the Indexer.

Thank you for your help.
Now waiting for a release.

Regards.

**Comment by harshit-splunk** [comment]
hey @npeyrussie, Glad to hear you got it working. We are planning for release soon.
> applying the commit diff to splunk-kubernetes-logging/templates/secret.yaml file.

Can you please clarify which commit diff you are talking about?

**Comment by npeyrussie** [comment]
I applied the diff from the commit you linked me : https://github.com/splunk/splunk-connect-for-kubernetes/pull/725/commits/840a05d5b296b58e43f02449bfdb24ca4bfd6e30

But only for the file `helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/secret.yaml`
Since we are only pushing logs to Splunk.

Sorry, I do not know how to directly link to a commit file on Github...

Regards.

**Comment by harshit-splunk** [comment]
@npeyrussie, that change was unnecessary so I have dropped the change. If you provided the certificate as a multiline string (use `|` operator in yaml file) that you have already done, it will work correctly. Can you please once again validate without applying diff?

**Comment by npeyrussie** [comment]
OK, it works... 
But I do not understand why, I am a bit confused.
I do no understand why it did not worked before...

**Comment by harshit-splunk** [comment]
Closing the issue.
Even I am not sure why it wasn't working earlier. Let us know if you find something.

**Comment by npeyrussie** [comment]
OK, no problem.
Thank you for your help.
